src/common/cockpitconf.c

```diff
@@ -21,92 +21,164 @@
 
 #include "cockpitconf.h"
 
-#include "cockpithash.h"
+#include <assert.h>
+#include <ctype.h>
+#include <err.h>
+#include <errno.h>
+#include <libgen.h>
+#include <limits.h>
+#include <regex.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* #define DEBUG 1 */
+#if DEBUG
+#define debug(fmt, ...) (fprintf (stderr, "cockpitconf: " fmt "\n", ##__VA_ARGS__))
+#else
+#define debug(...)
+#endif
+
+
+typedef struct Entry {
+    char *section;
+    char *key;
+    char *value;
+    char **strv_cache;
+    char strv_delimiter;
+    struct Entry *next;
+} Entry;
+
+static bool cockpit_conf_loaded = false;
+static Entry *cockpit_conf = NULL;
+
+const char *cockpit_config_file = "cockpit.conf";
+const char *cockpit_config_dirs[] = { PACKAGE_SYSCONF_DIR, NULL };
 
-#include "errno.h"
-
-static GHashTable *cockpit_conf = NULL;
-static GHashTable *cached_strvs = NULL;
-
-const gchar *cockpit_config_file = "cockpit.conf";
-const gchar *cockpit_config_dirs[] = { PACKAGE_SYSCONF_DIR, NULL };
+static void
+regcomp_errx (regex_t *preg, const char *regex, int cflags)
+{
+  int ret = regcomp (preg, regex, cflags);
+  if (ret != 0)
+    {
+      char err[1024];
+      regerror (ret, preg, err, sizeof (err));
+      errx (1, "failed to compile regular expression: %s\n", err);
+    }
+}
```

Comment:
This was for development. I could take this out now and replace with a simple `assert()`.

Reply:
Done.